morphir/toolkit/core/src/org/finos/morphir/ir/internal/TransformType.scala

```diff
@@ -0,0 +1,77 @@
+package org.finos.morphir.ir.internal
+
+import zio.Chunk
+import zio.prelude.fx.ZPure
+import org.finos.morphir.ir.Type.Type
+
+trait TransformType[T, A] extends Transform[T] {
```

Comment:
On a whole these are sufficient for rewrite operations but don't help if you want to transform into a brand new structure.
We need a `Parser` for starting with one value and producing the other.
We need a `Transpiler` for going from Morphir to Target IR/AST.
We need a `Writer` for writing to a destination type.